CMakeLists.txt

```diff
@@ -555,29 +555,10 @@ find_package (ZLIB REQUIRED)
 include_directories(${ZLIB_INCLUDE_DIRS})
 target_link_libraries (gerbera ${ZLIB_LIBRARIES})
 
-if(WITH_GOOGLETEST)
-    find_package(GTest)
-    if(GTEST_FOUND)
-        include(CTest)
-        set(gerberatestFILES
-            test/dictionary_test.cc
-            test/runtime_test.cc)
-
-        # Prevent GoogleTest from overriding our compiler/linker options
-        # when building with Visual Studio
-        set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-        include_directories(${GTEST_INCLUDE_DIRS})
-
-        # Trivial example using gtest and gmock
-        add_executable(gerberatest ${gerberatestFILES} $<TARGET_OBJECTS:libgerbera>)
-        target_include_directories(gerberatest PRIVATE "${CMAKE_SOURCE_DIR}/src")
-        target_include_directories(gerberatest PRIVATE "${CMAKE_SOURCE_DIR}/test")
-        define_file_path_for_sources(gerberatest)
-
-        get_target_property(configuredLibs gerbera INTERFACE_LINK_LIBRARIES)
-        target_link_libraries(gerberatest gtest gmock_main ${configuredLibs})
-        add_test(NAME example_test COMMAND gerberatest)
-    endif()
+if(WITH_TESTS)
```

Comment:
This **if** surrounds the sub-directory `/test`.  All `CMakeLists.txt` are children.  This allows the `/test` directory to be a self-contained system.